### PR TITLE
Pass in Admin connection to Script generator

### PR DESCRIPTION
**Pull request by janboll**
Need to pass in admin permission to Script Generator. Terraform-resources allows it to change the username of the admin connection. Need to pass the admin connection in to be able to create a managed user

### CODE REVIEW

File: reconcile/database_access_manager.py [hunk 1]
```diff
@@ -439,42 +442,52 @@ def _create_database_connection_parameter(
...
         user_secret_name,
         allow_not_found=True,
     )
+    admin_secret = oc.get(
+        namespace_name,
+        "Secret",
+        admin_secret_name,
+        allow_not_found=False,
+    )
+    print(f"admin_secret: {admin_secret}")
```
Comment (hemslo) [inline reconcile/database_access_manager.py:462]: debug?

File: reconcile/database_access_manager.py [hunk 2]
```diff
@@ -439,42 +442,52 @@ def _create_database_connection_parameter(
     oc: OCClient,
     admin_secret_name: str,
     user_secret_name: str,
-) -> DatabaseConnectionParameters:
+) -> dict[str, DatabaseConnectionParameters]:
     def _decode_secret_value(value: str) -> str:
         return base64.b64decode(value).decode("utf-8")
 
+    print("creating database connection parameter")
```
Comment (rporres) [inline reconcile/database_access_manager.py:449]: print? shouldn't this be done via `logging`?

File: reconcile/database_access_manager.py [hunk 3]
```diff
@@ -439,42 +442,52 @@ def _create_database_connection_parameter(
     oc: OCClient,
     admin_secret_name: str,
     user_secret_name: str,
-) -> DatabaseConnectionParameters:
+) -> dict[str, DatabaseConnectionParameters]:
```
Comment (hemslo) [inline reconcile/database_access_manager.py:445]: Another way is to create a new dataclass with 2 fields, or  typed dict to limit keys.

File: reconcile/database_access_manager.py (hunk 1 above)
Comment (janboll) [reply reconcile/database_access_manager.py]: :broom:

File: reconcile/database_access_manager.py (hunk 2 above)
Comment (janboll) [reply reconcile/database_access_manager.py]: :broom:

File: reconcile/database_access_manager.py (hunk 3 above)
Comment (janboll) [reply reconcile/database_access_manager.py:445]: Considered it, but since it's internal and just used right below it I omited that thought

**Review by hemslo — APPROVED** [review]
LGTM!

File: reconcile/database_access_manager.py (hunk 3 above)
Comment (hemslo) [reply reconcile/database_access_manager.py:445]: just for type check, still can be used as normal dict https://docs.python.org/3.8/library/typing.html#typing.TypedDict

**Pull request merged**